src/Common/src/TypeSystem/Common/TypeHashingAlgorithms.cs

```diff
@@ -138,11 +138,28 @@ public static int ComputeGenericInstanceHashCode<ARG>(int genericDefinitionHashC
         /// <param name="typeHashcode">Hashcode of the type that owns the method</param>
         /// <param name="nameOrNameAndGenericArgumentsHashcode">Hashcode of either the name of the method (for non-generic methods) or the GenericInstanceHashCode of the name+generic arguments of the method.</param>
         /// <returns></returns>
-        public static int ComputeMethodHashcode(int typeHashcode, int nameOrNameAndGenericArgumentsHashcode)
+        public static int ComputeMethodHashCode(int typeHashcode, int nameOrNameAndGenericArgumentsHashcode)
         {
             // TODO! This hash combining function isn't good, but it matches logic used in the past
             // consider changing to a better combining function once all uses use this function
             return typeHashcode ^ nameOrNameAndGenericArgumentsHashcode;
         }
+
+        /// <summary>
+        /// Produce a hashcode for a generic signature variable
+        /// </summary>
+        /// <param name="index">zero based index</param>
+        /// <param name="method">true if the signature variable describes a method</param>
+        public static int ComputeSignatureVariableHashCode(int index, bool method)
+        {
+            if (method)
+            {
+                return index * 0x7822381 + 0x54872645;
```

Comment:
Well, I didn't get those numbers, I'm just refactoring some code. In this case, you'd have to ask @jkotas about where he found the numbers to use here. However, in this case, the hash codes are not very useful as there are a highly limited number of possible variations in practice.